code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -352,9 +359,26 @@
 	R.my_atom = src.emag
 	R.add_reagent("beer2", 50)
 	src.emag.name = "Mickey Finn's Special Brew"
-	return
-
 
+/obj/item/weapon/robot_module/butler/recharge_consumable(var/mob/living/silicon/robot/R)
```

Comment:
I believe there is an OOP proc on items which gets called when recharging, can't remember exactly which.